### ISSUE TITLE
Docker does not start when I am signed in but does when I am not

### ISSUE THREAD

**Issue by bswilliamson** [docker/for-win#10896]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 7B0E70A2-2A7A-48B3-9346-AF07EFC553FF/20210412014256

### Actual behavior
Docker daemon does not start when I am signed in.

### Expected behavior
Docker daemon should start successfully when I am signed in.

### Information
  - Is it reproducible: yes
  - Is the problem new: yes
  - Did the problem appear with an update: yes
  - Windows Version: Windows 10 20H2 build 19042.867
  - Docker Desktop Version: 3.3.0 (62916)
  - Backend: WSL2 
  - Host: Win 10 developer VM running on Hyper-V

I am using Docker Desktop in a network environment that requires me to use a proxy server. If I am not currently signed in and then sign in it works but after a restart docker desktop fails to start the docker daemon (UI reports "stopped"). If I sign out it then starts, after which I can sign in again. 

I work in a large organisation and all traffic egresses via the same proxy which means that I must be signed in for image pulls to work due to the introduction of the new docker hub rate limit.

Possibly related to #10894.
 
### Steps to reproduce the behavior
1. Configure proxy settings in resources section
2. Login using "Sign in" button
3. Restart Windows

**Comment by stephen-turner** [comment]
I suspect your proxy software is interrupting the network connection between the host and the VM (where the Docker daemon runs). If so, it may require a configuration change in the proxy.

**Comment by bswilliamson** [comment]
Sorry I'm not sure what you mean. I'm not running any proxy software, I just have a proxy configured system wide and in the resources section of docker desktop. The thing is signing in after starting works fine, which means the proxy is working just fine, it just won't start if I have previously signed in. I should add the same happens if I kill all the docker services and restart the GUI.

**Comment by bswilliamson** [comment]
Doing a bit of digging in the logs I've found an IO timeout when connecting to hub.docker.com that could be the problem:
```
time="2021-04-12T04:07:47-07:00" level=error msg="failed to get entitlement: Get \"https://hub.docker.com/api/content/v1/entitlement/[REDACTED]/aggregate_features\": dial tcp 52.206.171.232:443: i/o timeout"
goroutine 2681 [running]:
github.com/docker/pinata/common/cmd/com.docker.backend/internal/features.(*FeatureManager).UpdateOnlineFeatures(0x275fe00, 0xc0000c0c30, 0x16, 0xc00033d0c0, 0x5, 0xc000190900, 0x87f, 0x0, 0x0)
	C:/workspaces/master/src/github.com/docker/pinata/common/cmd/com.docker.backend/internal/features/manager.go:77 +0x4f5
github.com/docker/pinata/common/cmd/com.docker.backend/internal/handlers.NewLoginInfo(0x1c9bfe8, 0xc0001e08c0, 0x48f0cd, 0x607429e5)
```
I've tested connectivity and I can connect to that host, it just fails due to lack of an auth header:
```
Invoke-WebRequest https://52.206.171.232:443/api/content/v1/entitlement/[REDACTED]/aggregate_features -Headers @{"Host"="hub.docker.com"}
Invoke-WebRequest : {"message":"required Authorization header not provided"}
At line:1 char:1
+ Invoke-WebRequest https://52.206.171.232:443/api/content/v1/entitleme ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```
To me this suggests that the proxy settings are not being used for this request.

**Comment by stephen-turner** [comment]
OK, thanks, we'll have a look.

**Comment by bswilliamson** [comment]
I have tried a few older versions including 3.1.0 and 3.2.2 and they work so this appears to be a regression in 3.3.0

**Comment by lavaude** [comment]
I do confirm the issue. I am also behind a corporate proxy.
If I am logged in on Docker hub, Docker for win is unusable. Once logged out, everything works fine again.
I also confirmed that this happened when upgrading to 3.3.0.

**Comment by stephen-turner** [comment]
Ok, thanks both of you, we're working on this now.

**Comment by mat007** [comment]
We’re thinking we have fixed this issue, here is [a private build with the fix](https://desktop-stage.docker.com/win/stable/amd64/63098/Docker%20Desktop%20Installer.exe) if you wanted to test it.
Thanks!

**Comment by mat007** [comment]
The fix has now been released with Docker Desktop 3.3.1. I’ll go ahead and close this issue. Thanks!

**Comment by bswilliamson** [comment]
I can confirm that this resolves the issue. Thanks for the speedy resolution!
